include/dataformats/ComponentRequest.hpp

```diff
@@ -22,16 +22,35 @@ namespace dataformats {
  */
 struct ComponentRequest
 {
+  /**
+   * @brief The current version of the ComponentRequest
+   */
+  static constexpr uint32_t s_component_request_version = 1; // NOLINT(build/unsigned)
+
+  /**
+   * @brief The version number of this ComponentRequest
+   */
+  uint32_t version{ s_component_request_version };
+
   GeoID component; ///< The Requested Component
 
   /**
    * @brief Start of the data collection window
    */
   timestamp_t window_begin{ TypeDefaults::s_invalid_timestamp };
+
   /**
    * @brief End of the data collection window
    */
   timestamp_t window_end{ TypeDefaults::s_invalid_timestamp };
+
+  ComponentRequest() {}
```

Comment:
Can we have the default constructor to set the timestamps and the GeoID to invalid?

Reply:
It does. Since it's not explicitly initializing the values, the default (invalid) initializers are used.

Reply:
True, sorry